exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/auth/DrillHttpSecurityHandlerProvider.java

```diff
@@ -54,11 +55,14 @@
   private final Map<String, DrillHttpConstraintSecurityHandler> securityHandlers =
       CaseInsensitiveMap.newHashMapWithExpectedSize(2);
 
+  private final BiConsumer<HttpServletRequest, HttpServletResponse> preHandleCallback;
+
   @SuppressWarnings("unchecked")
-  public DrillHttpSecurityHandlerProvider(DrillConfig config, DrillbitContext drillContext)
+  public DrillHttpSecurityHandlerProvider(DrillConfig config, DrillbitContext drillContext,
+                                          BiConsumer<HttpServletRequest, HttpServletResponse> preHandleCallback)
```

Comment:
Passing `BiConsumer` looks slightly complicated. Is it possible either to pass map with response headers or obtain headers from `DrillConfig` again and use it where needed?